frontend/packages/topology/src/components/export-app/ExportViewLogButton.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import { Button, Tooltip } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { resourcePath, history } from '@console/internal/components/utils';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { JobModel, PodModel } from '@console/internal/models';
+import { PodKind, JobKind } from '@console/internal/module/k8s';
+import { isModifiedEvent } from '@console/shared/src';
+import { EXPORT_JOB_NAME } from '../../const';
+
+interface ExportViewLogButtonProps {
+  namespace: string;
+  onViewLog?: () => void;
+}
+
+const ExportViewLogButton: React.FC<ExportViewLogButtonProps> = ({ namespace, onViewLog }) => {
+  const { t } = useTranslation();
+  const [job, jobLoaded] = useK8sWatchResource<JobKind>({
+    kind: JobModel.kind,
+    name: EXPORT_JOB_NAME,
+    namespace,
+    isList: false,
+  });
+
+  const podResource = React.useMemo(
+    () =>
+      jobLoaded &&
+      job?.metadata && {
+        kind: PodModel.kind,
+        isList: false,
+        namespace: job.metadata.namespace,
+        selector: job.spec.selector,
+      },
+    [job, jobLoaded],
+  );
+
+  const [podData, podLoaded] = useK8sWatchResource<PodKind>(podResource);
+
+  const path = React.useMemo(
+    () =>
+      podLoaded &&
+      podData?.kind === PodModel?.kind &&
+      podData?.metadata &&
+      `${resourcePath(PodModel.kind, podData?.metadata.name, podData?.metadata.namespace)}/logs`,
+
+    [podData, podLoaded],
+  );
```

Comment:
Using a memo typically isn't necessary for string computations as the value isn't heavy weight to compute.
Furthermore the `podData` object is a very broad object to have as a dependency and therefore will cause the path to be recomputed more often than needed since you really just need the name and namespace.

Reply:
removed Memo